### PR TITLE
[1809] Use interrupt page acknowledgements for rollover interrupt pages

### PR DESCRIPTION
**Pull request by FriedSock**
### Context
https://trello.com/c/ztZfy8sZ/1809-avoid-adding-acceptedrollover2022-user-state

### Changes proposed in this pull request

- Replaces the old mechanism for determining when to show interrupt pages based on user state
- Breaks apart the 2 rollover interrupt pages so they can be turned on independently 
- Calls a new endpoint on TTAPI for creating acknowledgements for interrupt pages per user/page type/recruitment cycle: https://github.com/DFE-Digital/teacher-training-api/pull/1942

### Guidance to review

Run locally in combination with https://github.com/DFE-Digital/teacher-training-api/pull/1942

- Check that the pages do not show up when rollover flags are off
- Create a new recruitment cycle on TTAPI (for the next year after the one you have set for current)
- Turn on `rollover.can_edit_current_and_next_cycles` and `rollover.show_next_cycle_allocation_recruitment_page` independently to check each page in isolation.
- Sign out and sign in again to verify that the acknowledgement is persisted

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Product Review

### CODE REVIEW

File: spec/controllers/application_controller_spec.rb
Comment (AbigailMcP) [inline spec/controllers/application_controller_spec.rb:216]: :)

**Comment by FriedSock** [comment]
[Avoid adding accepted_rollover_2022 user state](https://trello.com/c/ztZfy8sZ/1809-avoid-adding-acceptedrollover2022-user-state)

**Comment by FriedSock** [comment]
I would really prefer not to "fix" the thing codeclimate complains about. An elsif statement is simple. To make it stop complaining you either need to create more methods to have more indirection. or create a list of predicates and actions like this:
```
  def check_interrupt_redirects(use_redirect_back_to: true)
    # Codeclimate prefers this kind of madness
    redirects = [
      [-> { !user_from_session.accepted_terms? }, -> { redirect_to accept_terms_path }],
      [-> { user_from_session.aasm.current_state == :new }, -> { redirect_to transition_info_path }],
      [-> { show_rollover_page? }, -> { redirect_to rollover_path }],
      [-> { show_rollover_recruitment_page? }, -> { redirect_to rollover_recruitment_path }],
      [
        -> { use_redirect_back_to },
        lambda {
          redirect_to session[:redirect_back_to] if session[:redirect_back_to].present?
          session.delete(:redirect_back_to)
        },
      ],
    ]

    redirects.each do |predicate, action|
      next unless predicate.call

      action.call

      # rubocop thinks this is better than just "return"
      return nil
    end
  end
```

Which I think is overkill for a small elsif statement

**Comment by AbigailMcP** [comment]
> Which I think is overkill for a small elsif statement

this 🙏 

I'm very happy to leave as is!

**Review by AbigailMcP — COMMENTED** [review]
Happy with this! Just need to wait until the API one is in before merging this, so just holding off the 'Approve' until then in case we get merge happy!

File: app/controllers/application_controller.rb
Comment (gpeng) [inline app/controllers/application_controller.rb:232]: This looks good. Hard to get my head around sending the next cycle year but I'm not sure I've any better idea.

File: app/models/user.rb
Comment (gpeng) [inline app/models/user.rb:16]: I've started a ticket for this https://trello.com/c/6wQFnuhP/1443-migrate-users-to-transitioned if you want to remove the TODO. No problem if you want to leave it there for context though.

File: spec/support/features.rb
Comment (gpeng) [inline spec/support/features.rb:4]: We're definitely outgrowing features in settings. This is getting complicated 🤣

**Review by gpeng — COMMENTED** [review]
This is looking good 👍

File: app/controllers/application_controller.rb
Comment (FriedSock) [reply app/controllers/application_controller.rb:232]: I was thinking an alternative would be naming the pages per year, eg `rollover_2022`. But then it would mean we would need to keep updating the enum on TTAPI, or just drop validation altogether

**Review by AbigailMcP — APPROVED** [review]

**Pull request merged**
